### PR TITLE
[#950] improvment(common): Add a method in ConfigEntry called create to support configurations with no default value

### PR DESCRIPTION
**Pull request by qqqttt123**
### What changes were proposed in this pull request?
Some configuration options are necessary but don't have default value.

### Why are the changes needed?

Fix: #950 

### Does this PR introduce _any_ user-facing change?
Change some configuration options from `` to none

### How was this patch tested?
CI passed.

### CODE REVIEW

**Review by yuqi1129 — APPROVED** [review]
LGTM

File: common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java [hunk 1]
```diff
@@ -37,6 +37,7 @@ public class ConfigEntry<T> {
   @Getter private boolean isDeprecated;
 
   private boolean isOptional;
+  private boolean isNoDefault;
```
Comment (jerryshao) [inline common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java:40]: Can we use another name, `isNoDefault` seems a bit awkward.

File: common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java [hunk 2]
```diff
@@ -134,6 +140,20 @@ public ConfigEntry<Optional<T>> createWithOptional() {
     return conf;
   }
 
+  /**
+   * Creates a new ConfigEntry instance based on this configuration entry with no default value.
+   *
+   * @return A new ConfigEntry instance with no default value.
+   */
+  public ConfigEntry<T> createWithNoDefault() {
```
Comment (jerryshao) [inline common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java:148]: I think `create` is enough, no need to specify with "WithNoDefault".

File: server-common/src/main/java/com/datastrato/gravitino/server/auth/OAuthConfig.java [hunk 3]
```diff
@@ -33,7 +33,7 @@ public interface OAuthConfig extends Configs {
           .doc("The signing key of JWT when Gravitino uses OAuth as the authenticator")
           .version("0.3.0")
           .stringConf()
-          .createWithDefault("");
+          .createWithNoDefault();
```
Comment (jerryshao) [inline server-common/src/main/java/com/datastrato/gravitino/server/auth/OAuthConfig.java:36]: Maybe you should check all the configs not only here, but also in each catalogs, to change them if there're no default value.

File: common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java (hunk 2 above)
Comment (qqqttt123) [reply common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java:148]: OK, will fix.

File: common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java (hunk 1 above)
Comment (qqqttt123) [reply common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java:40]: How about `hasNoDefault`?

File: server-common/src/main/java/com/datastrato/gravitino/server/auth/OAuthConfig.java (hunk 3 above)
Comment (qqqttt123) [reply server-common/src/main/java/com/datastrato/gravitino/server/auth/OAuthConfig.java:36]: OK. I will modify them at the same time.

File: common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java (hunk 1 above)
Comment (jerryshao) [reply common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java:40]: Looks good.

File: server-common/src/main/java/com/datastrato/gravitino/server/auth/OAuthConfig.java (hunk 3 above)
Comment (qqqttt123) [reply server-common/src/main/java/com/datastrato/gravitino/server/auth/OAuthConfig.java]: Fixed.

File: common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java (hunk 1 above)
Comment (qqqttt123) [reply common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java]: Fixed.

**Review by jerryshao — APPROVED** [review]

**Comment by jerryshao** [comment]
@qqqttt123 can you please create a follow-up issue to address the validity of the value as we discussed offline?

**Pull request merged**